### PR TITLE
Log portfolio and application ids in jobs

### PR DESCRIPTION
**Pull request by jesse-dds**
https://ccpo.atlassian.net/browse/AT-5185

**Print any relevant errors/the response body for 400 responses from Azure (these are basically input validation errors in many cases)**

This seemed to be already covered. All of our API calls to azure seemed to follow this pattern:

```python
       try:
            sdk.apicall(...)
        except self.sdk.requests.exceptions.HTTPError:
            app.logger.error(f"...", exc_info=1)
            raise ConnectionException("connection error creating tenant")
```

The `exc_info` handles printing out the traceback, which includes info about the HTTP response.

**Stop printing full HTTP exchanges from Key Vault API calls**

This will be handled by this task https://ccpo.atlassian.net/browse/AT-5221 which will give us full control over what logging information we output. My attempts to retrieve the SDK's logger by name seemed to break the SDK's setup of its own logger and no information would output.

To reproduce this, set a test to `DEBUG` mode like so

```python
@pytest.mark.hybrid
def test_hybrid_create_application_job(session, csp, caplog):
    import logging

    caplog.set_level(logging.DEBUG)
    ...
```

And add `logging.getLogger("azure.core.pipeline.policies.http_logging_policy")` to our application's setup. I chose `make_app`. This line will cause the sdk to stop logging messages with "azure.core.pipeline.policies.http_logging_policy".

**Ensure linebreaks do not print as multiple log statements**

This will also be fixed by task https://ccpo.atlassian.net/browse/AT-5221

The only culprit I could find was the `azure.core.pipeline.policies.http_logging_policy` which seems to intentionally log each line of the HTTP response as a separate log message. Linebreaks do not necessarily print as multiple log statements. 

Here's a library doing it the way we want:
```
DEBUG [msal.token_cache] event={
    "response": {
        "access_token": "********",
        "expires_in": 3599,
        "expires_on": 1586534711,
        "ext_expires_in": 3599,
        "token_type": "Bearer"
    },
    "scope": [
        "https://vault.azure.net/.default"
    ]
}
```

Here are multiple logs per line:
```
INFO  [azure.core.pipeline.policies.http_logging_policy] Request method: 'GET'
INFO  [azure.core.pipeline.policies.http_logging_policy] Request headers:
INFO  [azure.core.pipeline.policies.http_logging_policy]     'Accept': 'application/json'
```

### CODE REVIEW

**Comment by jesse-dds** [comment]
I ignored tasks explicitly marked with `ignore_result=True`.

File: atat/jobs.py [hunk 1]
```diff
@@ -247,80 +271,101 @@ def make_initial_csp_data(portfolio):
...
-def dispatch_provision_portfolio(self):
+def dispatch_provision_portfolio(self: Task):
     """
     Iterate over portfolios with a corresponding State Machine that have not completed.
     """
+    portfolio_ids = []
     for portfolio_id in Portfolios.get_portfolios_pending_provisioning(pendulum.now()):
         provision_portfolio.delay(portfolio_id=portfolio_id)
+        portfolio_ids.append(portfolio_id)
+    return portfolio_ids
```
Comment (dandds) [inline atat/jobs.py:326]: Is it simpler/more readable to do:

```
portfolio_ids = Portfolios.get_portfolios_pending_provisioning(pendulum.now())
for portfolio_id in portfolio_ids:
   [etc]
return portfolio_ids
```
Genuine question. Probably 6 of one, half dozen of the other.

File: atat/jobs.py [hunk 2]
```diff
@@ -247,80 +271,101 @@ def make_initial_csp_data(portfolio):
...
     for application_id in Applications.get_applications_pending_creation():
         create_application.delay(application_id=application_id)
+        application_ids.append(application_id)
+    return application_ids
 
 
 @celery.task(bind=True)
-def dispatch_create_user(self):
+def dispatch_create_user(self: Task):
+    application_role_idss = []
```
Comment (dandds) [inline atat/jobs.py:340]: Maybe `application_role_id_groups`? 🤷‍♂️

File: atat/jobs.py (hunk 1 above)
Comment (jesse-dds) [reply atat/jobs.py:326]: That works, saves the append line too.

**Pull request merged**